django/db/migrations/state.py

```diff
@@ -352,7 +403,12 @@ def update_model_field_relation(
         remote_model_key = resolve_relation(model, *model_key)
         if remote_model_key[0] not in self.real_apps and remote_model_key in concretes:
             remote_model_key = concretes[remote_model_key]
-        self.relations[remote_model_key][model_key].append((field_name, field))
+        if field_name in self.models[model_key].fields:
```

Comment:
I think the lines below will be easier to understand if before this line, you define a variable equal to `self._relations[remote_model_key][model_key]`, or at least `self._relations[remote_model_key]`.